src/components/Form/FormField.tsx

```diff
@@ -0,0 +1,147 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as _ from 'lodash'
+
+import { UIComponent, customPropTypes, childrenExist } from '../../lib'
+import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
+import {
+  Extendable,
+  ReactChildren,
+  ShorthandValue,
+  ShorthandRenderFunction,
+} from '../../../types/utils'
+import Text from '../Text'
+
+export interface IFormFieldProps {
+  as?: any
+  children?: ReactChildren
+  className?: string
+  label?: ShorthandValue
+  renderLabel?: ShorthandRenderFunction
+  control?: React.ReactType<any>
```

Comment:
yes, agree, this should considered to be a shorthand.

Reply:
The logic for generating the control is a bit tricky. The control can be some component (we still don't have all components that can appear in the form implemented) or some HTML element (checkbox, button etc.). This is why this is just a type, and for consistency all properties available in the FormField are applied to the control element. What would be your proposal for using the shorthand in this case? Would be use the shorthand as one specific component and then all other components will always send the <Component ...props /> or some function and be responsible for applying all props inside that component?

Reply:
Here is the thing that I am concerned about - how we would be able to customize styles of this component? Yes, while it would be possible (with current approach) to pass `styles` to `Form.Field`, and those, if I've understood you correctly, will be passed to `content` element - for regular HTML element, however, we will have a problem that we would provide unknown `styles` attribute. This is the reason why I see this boundary should be clearly advertised - specifically, by providing the following semantics to the client: here is the `control` slot, please, provide all the necessary component props and style customizations there.

So, I would suggest to use `createHTMLInput` method from `factories.tsx` as a value for `content` - and even given that we have some problems with its implementation now, with this being fixed we should end up with correct implementation for `Form.Field` as well

Reply:
The control is converted to shorthand, and controlType prop is added that is the previous control prop. Depending on the controlType, different slot shorthand factories are created. @Bugaa92 I need your input whether this is the right approach. And yeah, I will need to add some custom tests for this prop, after we decide on the API used.